### PR TITLE
superlu-dist: add +parmetis variant.

### PR DESCRIPTION
**Pull request by jacobrking**
Expose ability to make parmetis an optional superlu-dist dependency to spack package management.

Parmetis as a dependency is enabled by default.

### CODE REVIEW

**Review by liuyangzhuan — APPROVED** [review]

**Review by eugeneswalker — DISMISSED** [review]

File: var/spack/repos/builtin/packages/superlu-dist/package.py
Comment (alalazo) [inline var/spack/repos/builtin/packages/superlu-dist/package.py]: Can you change the description to "Enable parmetis support" or something similar? Like this it seems the package is activating a vendored build, which is not the case.

**Review by alalazo — CHANGES_REQUESTED** [review]
Just a minor request to update the variant description

File: var/spack/repos/builtin/packages/superlu-dist/package.py
Comment (jacobrking) [reply var/spack/repos/builtin/packages/superlu-dist/package.py]: I copyed the description string from the package CMakeLists.txt but this makes sense to me. What do you think @liuyangzhuan ?

Comment (liuyangzhuan) [reply var/spack/repos/builtin/packages/superlu-dist/package.py]: @jacobrking  Yes "Enable parmetis library" makes more sense. I will change it in CMakeLists.txt. Could you change it for spack?

Comment (eugeneswalker) [reply var/spack/repos/builtin/packages/superlu-dist/package.py]: Just pushed these changes. Hope it is OK. If it is not please tell me and I won't do it again.

**Review by eugeneswalker — APPROVED** [review]

**Review by alalazo — APPROVED** [review]

**Comment by eugeneswalker** [comment]
@spackbot run pipeline - previous jobs were failing because of read timeouts, seem spurious, perhaps we'll have better luck this time

**Comment by eugeneswalker** [comment]
@spackbot run pipeline down to one job failing because of a read timeout, the other previously failing jobs succeeded on retry

File: var/spack/repos/builtin/packages/superlu-dist/package.py
Comment (jacobrking) [reply var/spack/repos/builtin/packages/superlu-dist/package.py]: Thank you for pushing the change. I'm at a conference this week so I'm slow to get to it.

**Comment by alalazo** [comment]
@spackbot run pipeline

**Comment by alalazo** [comment]
Apparently we are always getting the same fetch error here

**Comment by eugeneswalker** [comment]
@spackbot run pipeline

**Comment by eugeneswalker** [comment]
@spackbot run pipeline

**Comment by eugeneswalker** [comment]
All CI is green, should we wait for more review or is this ready to merge?

**Comment by jacobrking** [comment]
@liuyangzhuan Do you think we need more approvals? I just left the auto-populated reviewers when I made the PR. From my perspective this is ready to go.

**Comment by liuyangzhuan** [comment]
> @liuyangzhuan Do you think we need more approvals? I just left the auto-populated reviewers when I made the PR. From my perspective this is ready to go.
@jacobrking @eugeneswalker I think this is good to go. No more reviews is really needed.

**Pull request merged**
